dataland-frontend/src/utils/DataOwnerUtils.ts

```diff
@@ -0,0 +1,33 @@
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import type Keycloak from "keycloak-js";
+import { ApiClientProvider } from "@/services/ApiClients";
+import { type AxiosError } from "axios";
+import { waitForAndReturnResolvedKeycloakPromise } from "@/utils/KeycloakUtils";
+
+/**
+ * Check if a user is data owner of a company
+ * @param companyId the dataland companyId of the company for which ownership should be checked
+ * @param keycloakPromiseGetter the getter-function which returns a Keycloak-Promise
+ * @returns a promise, which resolves to a boolean
+ */
+export async function isUserDataOwnerForCompany(
+  companyId?: string,
+  keycloakPromiseGetter?: () => Promise<Keycloak>,
+): Promise<boolean> {
+  if (keycloakPromiseGetter && companyId) {
+    const resolvedKeycloakPromise = await waitForAndReturnResolvedKeycloakPromise(keycloakPromiseGetter);
+    const userId = resolvedKeycloakPromise?.idTokenParsed?.sub;
+    try {
+      await new ApiClientProvider(
+        assertDefined(keycloakPromiseGetter)(),
```

Comment:
And why do you re-run the keycloak-promise-getter here.
You already have the resolved keycloak promise in `const resolvedKeycloakPromimse`.
You can directly use that

Reply:
I tried but then i got an error message: ": Argument of type 'Keycloak' is not assignable to parameter of type 'Promise<Keycloak>'.   Type 'Keycloak' is missing the following properties from type 'Promise<Keycloak>': then, catch, finally,"
The ApiClientProvider expects a Promise<Keycloak>, but resolvedKeyCloakPromise is of type Keycloak and therefore i decided to just use the keycloakPromiseGetter